app/decorators/user_decorator.rb

```diff
@@ -1,9 +1,9 @@
 class UserDecorator < ApplicationDecorator
   delegate_all
 
-  def proposal_url(proposal)
+  def proposal_url(proposal, force_public_url=false)
     event = proposal.event
-    if model.staff_for? event
+    if model.staff_for?(event) && !force_public_url
```

Comment:
Yeah, I do think there were more references but we've simplified things now.  If so it would allow us to remove the need for the force_public_url flag, which would be nice.